service/collector/chart_config_resource.go

```diff
@@ -117,9 +118,14 @@ func (a *ChartConfigResource) Describe(ch chan<- *prometheus.Desc) error {
 }
 
 func convertToTime(datetime string) (time.Time, error) {
-	layout := "2006-01-02T15:04:05.000Z"
-	t, err := time.Parse(layout, datetime)
+	layout := "2006-01-02T15:04:05"
 
+	split := strings.Split(datetime, ".")
+	if len(split) == 0 {
+		return time.Time{}, microerror.Maskf(invalidExecutionError, "'%#v' must have at least one item in order to collect metrics for the cordon expiration", datetime)
```

Comment:
We had that a couple of times now and I would be interested why you chose `'%#v'` here? I am really curious at this point. 😄

Reply:
I copied from cert-operator and didn't care that much to change of it :(, I would convert it to `%#q` now.

Reply:
No worries. I understand. Let's change it in `cert-operator` as well maybe, as this project is not much worked on nowadays.